src/Components/Components/src/Forms/EditForm.cs

```diff
@@ -0,0 +1,139 @@
+        /// A callback that will be invoked when the form is submitted and the
+        /// <see cref="EditContext"/> is determined to be valid.
+        /// </summary>
+        [Parameter] EventCallback<EditContext> OnValidSubmit { get; set; }
+
+        /// <summary>
+        /// A callback that will be invoked when the form is submitted and the
+        /// <see cref="EditContext"/> is determined to be invalid.
+        /// </summary>
+        [Parameter] EventCallback<EditContext> OnInvalidSubmit { get; set; }
+
+        /// <inheritdoc />
+        protected override void OnParametersSet()
+        {
+            if ((EditContext == null) == (Model == null))
+            {
+                throw new InvalidOperationException($"{nameof(EditForm)} requires a {nameof(Model)} " +
+                    $"parameter, or an {nameof(EditContext)} parameter, but not both.");
+            }
+
+            // If you're using OnSubmit, it becomes your responsibility to trigger validation manually
+            // (e.g., so you can display a "pending" state in the UI). In that case you don't want the
+            // system to trigger a second validation implicitly, so don't combine it with the simplified
+            // OnValidSubmit/OnInvalidSubmit handlers.
+            if (OnSubmit.HasDelegate && (OnValidSubmit.HasDelegate || OnInvalidSubmit.HasDelegate))
+            {
+                throw new InvalidOperationException($"When supplying an {nameof(OnSubmit)} parameter to " +
+                    $"{nameof(EditForm)}, do not also supply {nameof(OnValidSubmit)} or {nameof(OnInvalidSubmit)}.");
+            }
+
+            // Update _fixedEditContext if we don't have one yet, or if they are supplying a
+            // potentially new EditContext, or if they are supplying a different Model
+            if (_fixedEditContext == null || EditContext != null || Model != _fixedEditContext.Model)
+            {
+                _fixedEditContext = EditContext ?? new EditContext(Model);
+            }
+        }
+
+        /// <inheritdoc />
+        protected override void BuildRenderTree(RenderTreeBuilder builder)
+        {
+            base.BuildRenderTree(builder);
+
+            // If _fixedEditContext changes, tear down and recreate all descendants.
+            // This is so we can safely use the IsFixed optimization on CascadingValue,
+            // optimizing for the common case where _fixedEditContext never changes.
+            builder.OpenRegion(_fixedEditContext.GetHashCode());
+
+            builder.OpenElement(0, "form");
+            builder.AddAttribute(1, "onsubmit", _handleSubmitDelegate);
+            builder.OpenComponent<CascadingValue<EditContext>>(2);
+            builder.AddAttribute(3, "IsFixed", true);
+            builder.AddAttribute(4, "Value", _fixedEditContext);
+            builder.AddAttribute(5, RenderTreeBuilder.ChildContent, ChildContent?.Invoke(_fixedEditContext));
+            builder.CloseComponent();
```

Comment:
I don't see a `method` attribute available in here. Don't we plan to support `post` method?

Reply:
The important use case is forms that are handled via an `onsubmit` handler, not ones that are directly posted to a server. As such, the `method` isn't applicable in normal use cases.

We are planning to allow arbitrary attribute pass-through at some point (a.k.a. "splatting"), at which point people could set a `method` or any other attribute if they wanted. We probably wouldn't want to put in any special-case support for `method` before then, because the use case is not major.